test/eigen.jl

```diff
@@ -75,24 +75,36 @@ using StaticArrays, Test, LinearAlgebra
         @test vals::SVector ≈ sort(m_d)
         @test eigvals(m_c) ≈ sort(m_d)
         @test eigvals(Hermitian(m_c)) ≈ sort(m_d)
+    end
 
-        # issue #523
-        for (i, j) in ((1, 2), (2, 1)), uplo in (:U, :L)
-            A = SMatrix{2,2,Float64}((i, 0, 0, j))
-            E = eigen(Symmetric(A, uplo))
-            @test eigvecs(E) * SDiagonal(eigvals(E)) * eigvecs(E)' ≈ A
-        end
-
-        m1_a = randn(2,2)
-        m1_a = m1_a*m1_a'
-        m1 = SMatrix{2,2}(m1_a)
-        m2_a = randn(2,2)
-        m2_a = m2_a*m2_a'
-        m2 = SMatrix{2,2}(m2_a)
-        @test (@inferred_maybe_allow SVector{2,ComplexF64} eigvals(m1, m2)) ≈ eigvals(m1_a, m2_a)
-        @test (@inferred_maybe_allow SVector{2,ComplexF64} eigvals(Symmetric(m1), Symmetric(m2))) ≈ eigvals(Symmetric(m1_a), Symmetric(m2_a))
+    # issue #523
+    @testset "2×2 degenerate cases" for (i, j) in ((1 , 1), (1, 2), (2, 1)), uplo in (:U, :L)
+        fmin = floatmin(Float64)
+        pfmin = prevfloat(fmin)
+        nfmin = nextfloat(fmin)
+        A = SMatrix{2,2,Float64}((i, 0, 0, j))
+        E = eigen(Symmetric(A, uplo))
+        @test eigvecs(E) * SDiagonal(eigvals(E)) * eigvecs(E)' ≈ A
+        A = SMatrix{2,2,Float64}((i, pfmin, pfmin, j))
+        E = eigen(Symmetric(A, uplo))
+        @test eigvecs(E) * SDiagonal(eigvals(E)) * eigvecs(E)' ≈ A
```

Comment:
Cool, so can you add some comment about exactly which cases we are straddling here by examining the boundary between `pfmin` vs `nfmin`?  We seem to have have `x^2 == 0` for both of those and `i ± x == 1.0`, `i*j  ± x == i*j` etc...

Reply:
I'd love to merge this, but I'm unsure whether these tests are testing the right thing. Do you have some thoughts on the above query?

Reply:
I looked into it again.  You were right, they were testing the wrong thing.  When I tested the right thing it wasn't working properly.  I put some time in to make it work and simplify things, mostly by using the hypot function which already has a lot of work put into making it accurate.